File: src/AdventureWorks.Application/DataEngine/Production/ProductPhoto/Commands/Queries/GetProductPhotos/ProductPhotoLookupModel.cs
using System;
using AutoMapper;
using AdventureWorks.Application.Interfaces.Mapping;
using AdventureWorks.BaseDomain.Entities.Production;
using System.Collections.Generic;
using AdventureWorks.Common.Extensions;
using AdventureWorks.BaseDomain.Enums;
using AdventureWorks.BaseDomain.Interfaces;
using AdventureWorks.BaseDomain.CustomTypes;

namespace AdventureWorks.Application.DataEngine.Production.ProductPhoto.Queries.GetProductPhotos
{
    public partial class ProductPhotoLookupModel : IHaveCustomMapping
    {
        public int ProductPhotoID { get; set; }
        public byte[] ThumbNailPhoto { get; set; }
        public string ThumbnailPhotoFileName { get; set; }
        public byte[] LargePhoto { get; set; }
        public string LargePhotoFileName { get; set; }
        public DateTime ModifiedDate { get; set; }

        #region Navigation Properties

        // Uncomment the following line if you need child navigation properties (can cause a poor performance)
        /*  */
        // Uncomment the following line if you need parent navigation properties
        /*  */
        #endregion

        public void CreateMappings(Profile configuration)
        {
            configuration.CreateMap<Domain.Entities.Production.ProductPhoto, ProductPhotoLookupModel>().IgnoreMissingDestinationMembers().CompensateWithDestinationValues();
            configuration.CreateMap<ProductPhotoLookupModel, BaseProductPhoto>().IgnoreMissingDestinationMembers();
        }
    }
}

File: src/AdventureWorks.Web.Api/ClientApp/src/app/models/data/entities/production/product-photo/product-photo-lookup-model.ts
import { IProductProductPhotoLookupModel } from 'app/models/data/entities/production/product-product-photo/product-product-photo-lookup-model';

export interface IProductPhotoLookupModel {
  productPhotoID: number;
  thumbNailPhoto: Int8Array;
  thumbnailPhotoFileName: string;
  largePhoto: Int8Array;
  largePhotoFileName: string;
  modifiedDate: Date | string;

  // Uncomment the following line if you need child navigation properties (can cause a poor performance)
  /*  */
  // Uncomment the following line if you need parent navigation properties
  /*  */
}

export class ProductPhotoLookupModel implements IProductPhotoLookupModel {
  productPhotoID: number;
  thumbNailPhoto: Int8Array;
  thumbnailPhotoFileName: string;
  largePhoto: Int8Array;
  largePhotoFileName: string;
  modifiedDate: Date | string;

  // Uncomment the following line if you need child navigation properties (can cause a poor performance)
  /*  */
  // Uncomment the following line if you need parent navigation properties
  /*  */

  constructor(init?: Partial<ProductPhotoLookupModel>) {
    if (!!init) {
      Object.assign<ProductPhotoLookupModel, Partial<ProductPhotoLookupModel>>(this, init);
    }
  }
}
